src/handler.rs

```diff
@@ -165,15 +178,36 @@ fn keystore_handler (keystore: &mut LedgerKeyStore, request: KeyStoreRequest) ->
             let account = keystore.borrow_account(&hash160)?;
             let drv_path = DerivationPath::from_str(&path).unwrap();
             let ledger_cap = account.extended_privkey(drv_path.as_ref())?;
-            if recoverable {
-                let signature = ledger_cap.begin_sign_recoverable(target_to_annotated_transaction(*target))?;
-                Ok(PluginResponse::Bytes(JsonBytes::from_vec(signature)))
-            } else {
-                Ok(PluginResponse::Error(JsonrpcError {
-                    code: 0,
-                    message: String::from("Non recoverable signing not supported"),
-                    data: None,
-                }))
+            let sign_msg = |(msg, display_hex)| -> Result <_, LedgerKeyStoreError> {
+                let magic_string = String::from("Nervos Message:");
```

Comment:
So the magic bytes should be added upstream. And then we need to check to see if the magic bytes exist in the message we recieve. If they don't, then we assume that the message is actually a message-hash, and we send `apdu_sign_message_hash`. If the bytes do exist, we send `apdu_sign_message`.

Reply:
We receive the payload without the magic_string. Though the hash of the payload we receive (`message` field) has been calculated with the magic_string prepended. (https://github.com/TheWaWaR/ckb-cli/blob/ledger-support/src/subcommands/util.rs#L363-L375)

About signing message-hash, its not implemented in this, I will push another commit for that.